contracts/YakVault.sol

```diff
@@ -177,41 +150,76 @@ contract YakVault is YakERC20, Ownable {
      */
     function removeStrategy(address strategy) public onlyOwner {
         require(strategy != activeStrategy, "YakVault::removeStrategy, cannot remove activeStrategy");
-        require(strategy != address(depositToken), "YakVault::removeStrategy, cannot remove deposit token");
-        require(supportedDepositTokens[strategy] == true, "YakVault::removeStrategy, not supported");
-        _revokeApproval(address(depositToken), strategy);
-        supportedDepositTokens[strategy] = false;
-        for (uint i = 0; i < supportedStrategies.length; i++) {
-            if (strategy == supportedStrategies[i]) {
-                supportedStrategies[i] = supportedStrategies[supportedStrategies.length - 1];
-                supportedStrategies.pop();
-                break;
-            }
-        }
+        require(supportedStrategies.contains(strategy) == true, "YakVault::removeStrategy, not supported");
+        require(getDeployedBalance(strategy) == 0, "YakVault::cannot remove strategy with funds");
+        depositToken.safeApprove(strategy, 0);
+        supportedStrategies.remove(strategy);
         emit RemoveStrategy(strategy);
     }
 
     /**
      * @notice Owner method for removing funds from strategy (to rebalance, typically)
      * @param strategy address
      * @param amount deposit tokens
+     * @param withdrawPercentageBips percentage to withdraw from strategy, pass 0 to use amount parameter
      */
-    function withdrawFromStrategy(address strategy, uint amount) public onlyOwner {
-        uint balanceBefore = depositToken.balanceOf(address(this));
-        uint strategyShares = YakStrategy(strategy).getSharesForDepositTokens(amount);
-        YakStrategy(strategy).withdraw(strategyShares);
-        uint balanceAfter = depositToken.balanceOf(address(this));
+    function withdrawFromStrategy(
+        address strategy,
+        uint256 amount,
+        uint256 withdrawPercentageBips
+    ) public onlyOwner {
+        _withdrawFromStrategy(strategy, amount, withdrawPercentageBips);
+    }
+
+    function _withdrawFromStrategy(
+        address strategy,
+        uint256 amount,
+        uint256 withdrawPercentageBips
+    ) private {
+        require(withdrawPercentageBips <= BIPS_DIVISOR, "YakVault::withdrawFromStrategy invalid percentage");
+        uint256 balanceBefore = depositToken.balanceOf(address(this));
+        uint256 withdrawalStrategyShares = 0;
+        if (withdrawPercentageBips > 0) {
```

Comment:
here we either withdraw % or amount (denominated in deposit tokens). We could fully separate this functionality to avoid confusion 

* function _withdrawShares(shares)
* function _withdrawPercent(percent)
* function _withdrawDepositTokens(depositTokenAmount)

Reply:
I agree!

_withdrawShares isn't required internally right now and I don't see a reason to add an owner method for it, so I didn't add it. Keep in mind _withdrawFromStrategy is private and only used internally and via public owner protected method. User facing we still have the standard withdraw(share amount) method.
Kept _withdrawFromStrategy for deposit token amount and added _withdrawPercentageFromStrategy

Did the same for the owner only depositToStrategy methods.